src/doc/src/faq.md

```diff
@@ -259,3 +259,40 @@ Some issues we've seen historically which can cause crates to get rebuilt are:
 If after trying to debug your issue, however, you're still running into problems
 then feel free to [open an
 issue](https://github.com/rust-lang/cargo/issues/new)!
+
+### Why does Cargo.lock conflict?
+
+`Cargo.lock` contains exact information about your dependencies. It should be 
+generated and maintained by Cargo and should not be edited manually. If there is 
+a dependency conflict caused by changing the content of `Cargo.lock` when editing 
+or merging branches, it is recommended to try to run `cargo generate-lockfile` to 
+solve it.
```

Comment:
I personally wouldn't recommend this. `cargo generate-lockfile` will update all dependencies, which is usually not what is wanted.

I think most conflicts can be trivially resolved manually. The format is not that complicated, and editing manually shouldn't be such a concern.

I would say that if there are non-trivial conflicts, you can reset all "yours" changes, fix all other conflicts in the branch, and then run some cargo command (like `cargo tree` or `cargo check`), which should re-update the lockfile with your own local changes. If you previously ran some `cargo update` commands in your branch, you can re-run them that this time.